backend/controllers/controller_helm_applications.go

```diff
@@ -0,0 +1,83 @@
+package controllers
+
+import (
+	"fmt"
+	"github.com/ZPI-2024-25/KubernetesAccessManager/helm"
+	"github.com/ZPI-2024-25/KubernetesAccessManager/models"
+	"net/http"
+)
+
+func GetHelmReleaseController(w http.ResponseWriter, r *http.Request) {
+	handleHelmOperation(w, r, models.Read, func(releaseName, namespace string) (interface{}, *models.ModelError) {
+		return helm.GetHelmRelease(releaseName, namespace)
+	})
+}
+
+func GetHelmReleaseHistoryController(w http.ResponseWriter, r *http.Request) {
+	handleHelmOperation(w, r, models.Update, func(releaseName, namespace string) (interface{}, *models.ModelError) {
+		return helm.GetHelmReleaseHistory(releaseName, namespace)
+	})
+}
+
+func ListHelmReleasesController(w http.ResponseWriter, r *http.Request) {
+	handleHelmOperation(w, r, models.List, func(releaseName, namespace string) (interface{}, *models.ModelError) {
+		return helm.ListHelmReleases(namespace)
+	})
+}
+
+func RollbackHelmReleaseController(w http.ResponseWriter, r *http.Request) {
+	handleHelmOperation(w, r, models.Update, func(releaseName, namespace string) (interface{}, *models.ModelError) {
+		var version models.ReleaseNameRollbackBody
+		if !decodeJSONBody(r, &version) {
+			return nil, &models.ModelError{Code: http.StatusBadRequest, Message: "Invalid request body"}
+		}
+		if err := checkVersion(version.Version); err != nil {
+			return nil, err
+		}
+
+		return helm.RollbackHelmRelease(releaseName, namespace, int(version.Version))
+	})
+}
+
+func UninstallHelmReleaseController(w http.ResponseWriter, r *http.Request) {
```

Comment:
Maybe not the best place to adress this issue, but when we have a chart that takes some time to uninstall, that endpoint waits until it is uninstalled. I think this should not be the case. Check if other endpoints also have this problem.
Example chart:
https://chatgpt.com/share/6731bbd3-b23c-800b-9ce8-4fdf47085f9b

Reply:
I modified `UninstallHelmReleaseController` to wait 10 seconds before sending an accepted response, then proceed with the uninstallation in the background. After verification, I will apply similar changes to `RollbackHelmReleaseController`.

API will need to be changed.

Reply:
I would change it to even shorter time - for example 5 sec. If API needs to be changed please create Jira ticket with explanation why, what and where should be changed